lexos/models/topword_model.py

```diff
@@ -0,0 +1,377 @@
+        Usually we report a word as an anomaly if the return value is smaller
+        than -1.96 or bigger than 1.96.
+        :param p1: the probability of a word's occurrence in a particular
+                   segment: Number of word occurrence in the segment /
+                   total word count in the segment
+        :param p2: the probability of a word's occurrence in all the segments
+                   Number of word occurrence in all the segment /
+                   total word count in all the segment
+        :param n1: the number of total words in this segment.
+        :param n2: the number of total words in this segment.
+        :return: the z-score indicates that the particular word in a particular
+                 segment is an anomaly or not.
+        """
+        # Trap possible empty inputs.
+        assert n1 > 0 and n2 > 0, SEG_NON_POSITIVE_MESSAGE
+
+        # Find the estimator of overall sample proportion.
+        p_hat = (p1 * n1 + p2 * n2) / (n1 + n2)
+        # Find the standard error.
+        standard_error = np.sqrt(p_hat * (1 - p_hat) * ((1 / n1) + (1 / n2)))
+
+        # Trap possible division by 0 error.
+        # TODO: Do we may need more complicate check here?
+        if math.isclose(standard_error, 0):
+            return 0.0
+        # If not division by 0, return the calculated z-score.
+        else:
+            return round((p1 - p2) / standard_error, 4)
+
+    @staticmethod
+    def _z_test_word_list(data_series_one: pd.Series,
+                          data_series_two: pd.Series) -> pd.Series:
+        """Run z-test on all the words of two input word lists.
+
+        :param data_series_one: a pandas series where data represents word
+                                counts and corresponding index are the words.
+        :param data_series_two: a pandas series where data represents word
+                                counts and corresponding index are the words.
+        :return: a panda series contains analysis result, where index are words
+                 and corresponding data is the z_score. And the panda series is
+                 sorted by z_score in descending order.
+        """
+
+        # Find sample population of the two input data set
+        n1 = data_series_one.values.sum()
+        n2 = data_series_two.values.sum()
+
+        def _z_test_for_one_word(word_index: int) -> float:
+            """Helper function that run z-test on one word.
+
+            :param word_index: index of the word.
+            :return: z-test score.
+            """
+            p1 = data_series_one[word_index] / n1
+            p2 = data_series_two[word_index] / n2
+            return TopwordModel._z_test(p1=p1, p2=p2, n1=n1, n2=n2)
+
+        # Perform the z-test to detect word anomalies.
+        full_word_score_dict = \
+            {word: _z_test_for_one_word(word_index=index)
+             for index, word in enumerate(data_series_one.index.values)}
```

Comment:
Do not work with index sir. You cannot assume that `data_series_one` and `data_series_two` are parallel arrays. the correct way is to combine series together and iterate through the words.

Untested code:
```python
# merge 2 data series into a 2 columns data frame
merged_data_series = pd.concat([data_series_one, data_series_two], axis="columns")

# iterate through the word count
full_word_score_dict = {
    word: TopwordModel._z_test(p1=count1/n1, p2=count2/n2, n1=n1, n2=n2)  \
    for word, [count1, count2] in emerged_data_series.iterrows()
} 
```

Reply:
remind me again, why is the `full_word_score_dict` not a panda series?

Reply:
Would You please add some documentation about why do we use dictionary instead of panda series.
like this:
```
We are using dict instead of panda series here, 
because this function requires `full_word_score_dict` to be sorted via
the absolute value of the value. 
So far, (panda 0.21) does not offer a decent series sortby option.
In the future we can implement it in out general functions.
But now, just for code clarity we will use dictionary as a temporary solution
```

Reply:
Yes boss. Once I fixed the parallel array problem and other issues, I'll go through the file and add more documentations, including this one.